portfolio_optimization/cvar_portfolio_optimization.ipynb

```diff
@@ -0,0 +1,1092 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "# CVaR Portfolio Optimization with cuOpt Python API\n",
+        "\n",
+        "This notebook demonstrates Conditional Value at Risk (CVaR) portfolio optimization using NVIDIA's cuOpt Python API with S&P 500 stock data. The approach is inspired by the MOSEK CVaR risk measure methodology but implemented using GPU-accelerated optimization.\n",
+        "\n",
+        "## Overview\n",
+        "\n",
+        "**Conditional Value at Risk (CVaR)** is a risk measure that quantifies the expected loss in the worst-case scenarios beyond a certain confidence level. It's particularly useful for portfolio optimization as it provides a coherent risk measure that captures tail risk.\n",
+        "\n",
+        "### CVaR Formulation\n",
+        "\n",
+        "The CVaR portfolio optimization problem can be formulated as:\n",
+        "\n",
+        "$$\n",
+        "\\begin{align}\n",
+        "\\text{maximize: } & \\mu^T w - \\lambda \\text{CVaR}_\\alpha(w) \\\\\n",
+        "\\text{subject to: } & \\mathbf{1}^T w = 1 \\\\\n",
+        "& w_i^{\\min} \\leq w_i \\leq w_i^{\\max}, \\quad i = 1, \\ldots, n\n",
+        "\\end{align}\n",
+        "$$\n",
+        "\n",
+        "Where:\n",
+        "- $w$ is the portfolio weight vector\n",
+        "- $\\mu$ is the expected return vector\n",
+        "- $\\lambda$ is the risk aversion parameter\n",
+        "- $\\text{CVaR}_\\alpha(w)$ is the Conditional Value at Risk at confidence level $\\alpha$\n",
+        "\n",
+        "### Data Source\n",
+        "We use S&P 500 stock data from `./cuFOLIO_portfolio_optimization/data/stock_data/sp500.csv` which contains historical price data for S&P 500 constituents.\n",
+        "\n",
+        "### Requirements\n",
+        "- **GPU**: NVIDIA GPU with CUDA support (recommended for optimal performance)\n",
+        "- **CUDA**: Version 11.x or 12.x\n",
+        "- **Python**: 3.8 or higher\n",
```

Comment:
Is Python 3.8 right?

Reply:
Nah, it should have been >= 3.10 <=3.13, great catch, let me update these